core/src/main/java/com/klaytn/caver/contract/Contract.java

```diff
@@ -0,0 +1,349 @@
+
+    /**
+     * The map where method name string and ContractMethod mapped.
+     */
+    Map<String, ContractMethod> methods;
+
+    /**
+     * The map where event name string and ContractEvent mapped.
+     */
+    Map<String, ContractEvent> events;
+
+    /**
+     * The ContractMethod instance related Contract's constructor.
+     */
+    ContractMethod constructor;
+
+    /**
+     * Creates a Contract instance.
+     * @param caver A Caver instance.
+     * @param abi A contract's ABI(Application Binary interface) json string.
+     * @throws IOException
+     */
+    public Contract(Caver caver, String abi) throws IOException{
+        this(caver, abi, null);
+    }
+
+    /**
+     * Creates a Contract instance.
+     * @param caver A Caver instance
+     * @param abi A contract's ABI(Application Binary Interface) json string.
+     * @param contractAddress
+     * @throws IOException
+     */
+    public Contract(Caver caver, String abi, String contractAddress) throws IOException{
+        setAbi(abi);
+        setCaver(caver);
+        setContractAddress(contractAddress);
+    }
+
+    /**
+     * Deploy a contract.
+     * It sets TransactionReceiptProcessor to PollingTransactionReceiptProcessor instance.
+     * @param deployParam A DeployParam instance.
+     * @param sendOptions A SendOption instance.
+     * @return Contract
+     * @throws IOException
+     * @throws TransactionException
+     */
+    public Contract deploy(ContractDeployParam deployParam, SendOptions sendOptions) throws IOException, TransactionException {
+        return deploy(deployParam, sendOptions, new PollingTransactionReceiptProcessor(caver, 1000, 15));
+    }
+
+    /**
+     * Deploy a contract
+     * @param deployParam A DeployParam instance.
+     * @param sendOptions A SendOption instance.
+     * @param processor A TransactionReceiptProcessor instance.
+     * @return Contract
+     * @throws IOException
+     * @throws TransactionException
+     */
+    public Contract deploy(ContractDeployParam deployParam, SendOptions sendOptions, TransactionReceiptProcessor processor) throws IOException, TransactionException {
+        List<Type> deployParams = deployParam.getDeployParams();
+        this.constructor.checkTypeValid(deployParams);
+
+        String encodedParams = ABI.encodeParameters(deployParams);
+        String input = deployParam.getBytecode() + encodedParams;
+
+        SmartContractDeploy smartContractDeploy = new SmartContractDeploy.Builder()
+                .setKlaytnCall(caver.rpc.klay)
+                .setFrom(sendOptions.getFrom())
+                .setInput(input)
+                .setCodeFormat(CodeFormat.EVM)
+                .setHumanReadable(false)
+                .setGas(sendOptions.getGas())
+                .build();
+
+        caver.wallet.sign(sendOptions.getFrom(), smartContractDeploy);
+
+        Bytes32 txHash = caver.rpc.klay.sendRawTransaction(smartContractDeploy.getRawTransaction()).send();
```

Comment:
This is just question. (not related with this PR)
Can the sendRawTransaction function take a signed transaction object as a parameter?

Reply:
Did "signed transaction object" means smartContractDeploy?
It is possible to pass the signed transaction object as a parameter.